diff --git a/client/src/components/Feed/Post.js b/client/src/components/Feed/Post.js
index 4ac65b36..0ab93a4a 100644
--- a/client/src/components/Feed/Post.js
+++ b/client/src/components/Feed/Post.js
@@ -200,7 +200,9 @@ const Post = ({
     }
     const currentLimit = limit.current;
     //limit.current = currentLimit * page;
-    limit.current = currentLimit * (page == 1 ? 2 : page); // a workaround to fix above line,
+    console.log(limit.current);
+    //limit.current = currentLimit * (page == 1 ? 2 : page); // a workaround to fix above line,
+    limit.current = currentLimit + 5; // a workaround to fix above line,
     // when page == 1 not expand any more,
     // to reproduce, use above line, create a post, add 10 comments, toggle show less/more several times
   };
@@ -273,6 +275,7 @@ const Post = ({
       postDelete(post);
     } else if (deleteModalVisibility === DELETE_MODAL_COMMENT) {
       deleteComment(comment);
+      loadComments();
     }
 
     setToDelete("");
